### PR TITLE
Reintroduce integration package generator

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Introduce a new integration package generator/transformer, which takes a partial integration package in apmpackage/apm and performs APM-specific transformations and validations:

 - Inject package version and Kibana version constraint
 - Inject common ingest pipelines
 - Check ilm_policy is present and correct (and fix the
   value in the sampled_traces)
 - Copy fields from traces to rum_traces

We no longer explicitly set `event.ingested` in our ingest pipelines; this is done by the common Fleet-installed pipeline, which is set as the final pipeline in all Fleet-install templates.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
~- [ ] Documentation has been updated~

## How to test these changes

1. Install the package
2. Check all index templates' ILM policy references are valid

## Related issues

Prereq for https://github.com/elastic/apm-server/issues/6425

### CODE REVIEW

File: apmpackage/cmd/genpackage/main.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+var (
+	outputDir = flag.String("o", "", "directory into which the package will be rendered (required)")
+	version   = flag.String("version", "", "integration package version (required)")
+)
+
+func generatePackage(pkgfs fs.FS, version *common.Version) error {
+	// Walk files, performing some APM-specific validations and transformations as we go.
+	//
+	// We assume the target destination does not yet exist.
+	if err := fs.WalkDir(pkgfs, ".", func(path string, d fs.DirEntry, err error) error {
```
Comment (marclop) [inline apmpackage/cmd/genpackage/main.go:42]: We can return here directly, no need to handle the error, right?

**Review by simitt — APPROVED** [review]
Briefly looked at the generated apmpackage - LGTM

**Review by marclop — APPROVED** [review]
LGTM, also nicer to have the checks in the generator itself than parsing things in the Makefile.

File: apmpackage/cmd/genpackage/main.go (hunk 1 above)
Comment (axw) [reply apmpackage/cmd/genpackage/main.go:42]: Indeed. I'll do that in my followup PR, in the interest of time.

**Comment by marclop** [comment]
Verified with 9bb5e4cdd and apmpackage `8.0.0-dev4`

**Pull request merged**